src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -84,6 +88,44 @@ public async Task<IActionResult> CreateOrUpdate(string name, [FromBody] Function
             }
         }
 
+        [HttpPost]
+        [Route("admin/warmup")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevel)]
+        [RequiresRunningHost]
+        public async Task<IActionResult> Warmup([FromServices] IScriptJobHost scriptHost, [FromServices] IScriptHostManager scriptHostManager, [FromServices] IHostIdProvider hostIdProvider)
+        {
+            // The function must be called warmup and should be of type warmup trigger
+            FunctionDescriptor function = scriptHost.GetFunctionOrNull(WarmUpConstants.FunctionName);
+            if (function != null && function.Metadata.FunctionContainsInputBinding(WarmUpConstants.WarmupTrigger))
+            {
+                try
+                {
+                    var hostStatus = new HostStatus
```

Comment:
nit: in HostController we're already newing one of these up and populating in the same way. Once you move this to that controller, we should reuse code.

Reply:
Now that I am only passing the hostState, there's not much to reuse.